src/Grpc.AspNetCore.Server/Internal/GrpcProtocolHelpers.cs

```diff
@@ -90,5 +93,71 @@ public static byte[] ParseBinaryHeader(string base64)
 
             return Convert.FromBase64String(decodable);
         }
+
+        internal static bool TryDecompressMessage(string compressionEncoding, byte[] messageData, out byte[] result)
+        {
+            switch (compressionEncoding)
+            {
+                case "gzip":
```

Comment:
Hardcoded. Add an API for new compression encodings? For example, https://docs.microsoft.com/en-us/aspnet/core/performance/response-compression?view=aspnetcore-2.2#custom-providers

Reply:
Add providers